### PR TITLE
feat(http): add custom method support to http_scrape source

### PR DESCRIPTION
**Pull request by zamazan4ik**
This PR partially resolves https://github.com/vectordotdev/vector/issues/14676

### Supported HTTP methods
I have enabled only GET and POST methods like the most reasonable. Other methods could be enabled quite easily later if will be a need in it (just extend corresponding enum and update the documentation). This is the reason, why `HttpMethod` enum was copied from `http` source - they support different set of HTTP methods and I see no need to support in both sources equal set of HTTP methods.

### CODE REVIEW

**Comment by spencergilbert** [comment]
Hey, thanks for the contribution. I think a cleaner way of doing this would be moving this complete [HttpMethod](https://github.com/vectordotdev/vector/blob/master/src/sources/http.rs#L36-L55) enum to a shared location and re-use it between both the server and client.

Rather than deriving the default, each could implement `default()` themselves to allow for the server to remain as `POST` and the client as `GET`.

What do you think?

**Comment by zamazan4ik** [comment]
> I think a cleaner way of doing this would be moving this complete [HttpMethod](https://github.com/vectordotdev/vector/blob/master/src/sources/http.rs#L36-L55) enum to a shared location and re-use it between both the server and client.

I explicitly copied `HttpMethod` since http client and server support different HttpMethod subsets. That's why they are separated.

> Rather than deriving the default, each could implement default() themselves to allow for the server to remain as POST and the client as GET.

I have just did it as it done in `http` source :) If you want this change, we need to refactor not only the new functionality, but also the old one. From my perspective (not so proficient Rust developer) I do not see a huge difference between deriving default and implementing `Default` trait manually.

File: src/sources/util/http_scrape.rs [hunk 1]
```diff
@@ -105,6 +106,7 @@ pub(crate) async fn http_scrape<
     inputs: GenericHttpScrapeInputs,
     context_builder: B,
     mut out: SourceSender,
+    http_method: Option<HttpMethod>,
```
Comment (neuronull) [inline src/sources/util/http_scrape.rs:109]: I think it is a little cleaner to have this not be an `Option`.
The function still resolves to one of the `HttpMethod` enum variants if `None` is passed in (there isn't different behavior taken in the `None` case).

Comment (zamazan4ik) [reply src/sources/util/http_scrape.rs:109]: So what do you suggest? If I will make just `HttpMethod`, I will need to pass `HttpMethod::Get` in `prometheus` source. Is it ok? If so, I will do it.

Comment (neuronull) [reply src/sources/util/http_scrape.rs:109]: > I will need to pass HttpMethod::Get in prometheus source

:+1: yes, this is what I'd suggest doing.

**Comment by neuronull** [comment]
This looks good to me. I added the integration tests since they were modified.

**Review by spencergilbert — APPROVED** [review]
Awesome! Thanks for the contribution!

**Review by neuronull — APPROVED** [review]

**Pull request merged**